### ISSUE TITLE
Parser rejects empty record types

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#95]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/95/) on Sourceforge by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-07 19:52:49.970000:

From JPCW.  Parser rejects:

```
types
  k1 ::
  k2 ::
  con :: b : bool
  mytype = k1 | k2 | con
```

It expects a non-empty sequence of fields in the record types.

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/95/#0b24) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-10 07:48:40.828000:
- **status**: accepted --> closed

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/95/#4656) by [joeycoleman](https://sourceforge.net/u/joeycoleman/profile/), 2013-10-10 07:48:41.140000:

Fixed in [e731b66019ab0c705768b97241750adf1014f97c]

Empty record types are now parsed correctly, and existing typechecker and pog tests pass...
